test/integration/agent-integration.test.ts

```diff
@@ -0,0 +1,44 @@
+import { describe } from 'mocha'
+import * as puppeteer from 'puppeteer'
+import AgentService from '../../src/services/agent-service'
+import Constants from '../../src/services/constants'
+import { agentJsFilename } from '../../src/utils/sdk-utils'
+import chai from '../support/chai'
+const expect = chai.expect
+
+// Use this instead of importing PercyAgent - we only want the browserified version
+declare var PercyAgent: any
+
+async function snapshot(page: puppeteer.Page, name: string, options: any = {}) {
+  await page.addScriptTag({path: agentJsFilename()})
+  return page.evaluate((name: string, options: any) => {
+    const percyAgentClient = new PercyAgent({ clientInfo: 'integration-microSDK' })
+    return percyAgentClient.snapshot(name, options)
+  }, name, options)
+}
+
+describe('Integration test', () => {
+  let browser: puppeteer.Browser | null = null
+  let page: puppeteer.Page | null = null
+
+  before(async () => {
+    browser = await puppeteer.launch({
+      headless: true,
+      args: [
+        '--disable-gpu',
+        '--single-process',
+      ],
+    })
+    page = await browser.newPage()
+  })
+
+  after(async () => {
+    browser!.close()
```

Comment:
@djones I don't love having to use `!` here and on lines 40 and 41, but I did not see a better way of getting Typescript to stop complaining about how `browser` and `page` could be `null`. Let me know if there is some Typescript magik that I'm missing.

Reply:
I think if you remove the two ` | null = null` bits from `let browser` and `let page` above you can remove all the `!`s

Reply:
Yes! The key for me is that I didn't realize that I can just skip assigning `null` in the declaration. 🤦🏼‍♀️